### DISCUSSION TITLE
Help with Start Gcode

### DISCUSSION THREAD

**Discussion by john-helyar** [mriscoc/Ender3V2S1#639] [Q&A]
Hi there i wonder if you can help please I have an Ender 3v2, and I was using Jyers Firmware but have now moved over to yours, but I need help with my start gcode

I did have it so when I started each print, it would do the bed levelling using my CR Touch, and it would save it and then start printing, but since moving to your firmware, it doesn't do it not sure why a copy of my code is below also I always have problem with ozzing so it would be great to be able to set it a slightly lower temp while it does the leveling if you can help that would be great.

Thank you in advance

``` Ender 3 Custom Start G-code
G92 E0 ; Reset Extruder
M140 S{material_bed_temperature_layer_0} ; start preheating the bed
M104 S{material_print_temperature_layer_0} ﻿T0 ; start preheating hotend
G28 ; Home all axes
M190 S{material_bed_temperature_layer_0} ; heat to Cura Bed setting
M109 S{material_print_temperature_layer_0} ﻿T0 ; heat to Cura Hotend
G29 ; Level Print Bed
M500 ; Save to EEPROM
C108 ; Close Mesh Viewer
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish```

**Comment by mriscoc** [comment]
For UBL you need to change `G29` to `G29 P1`
https://github.com/mriscoc/Ender3V2S1/wiki/Slicer-G-code-Scripts

**Reply by john-helyar** [reply]
Any advice? on the G-Code below

**Comment by john-helyar** [comment]
ok thats great thank you for that so simply change G29 to G29 P1 I assume the M500 still saves it.

How do I go about setting the temp to say 150C so its hot but not going to ooze out as even after it does the wipe down the side it sometimes starts the print with a nice blob on the end of the nozzle

So idealy just want it to heat up the bed to 60C nozzle to 150C level everything then go to the temps I have set in Cura 

as you can gather new to editing the startup Gcode

**Comment by john-helyar** [comment]
Ok so i came up with this not sure if it is correct I haven't tried it yet but if you can let me know what you think

```G92 E0 ; Reset Extruder
M140 S{material_bed_temperature_layer_0} ; start preheating the bed
M104 S{material_print_temperature_layer_0} ﻿T0 ; start preheating hotend
G28 ; Home all axes
M190 S{material_bed_temperature_layer_0} ; heat to Cura Bed setting
M104 S150 ; set temporary nozzle temp to prevent oozing during homing
G4 S10 ; allow partial nozzle warmup
G29 P1; Level Print Bed
M500 ; Save to EEPROM
C108 ; Close Mesh Viewer
M109 S{material_print_temperature_layer_0} ﻿T0 ; heat to Cura Hotend
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X0.1 Y20 Z0.3 F5000.0 ; Move to start position
G1 X0.1 Y200.0 Z0.3 F1500.0 E15 ; Draw the first line
G1 X0.4 Y200.0 Z0.3 F5000.0 ; Move to side a little
G1 X0.4 Y20 Z0.3 F1500.0 E30 ; Draw the second line
G92 E0 ; Reset Extruder
G1 Z2.0 F3000 ; Move Z Axis up little to prevent scratching of Heat Bed
G1 X5 Y20 Z0.3 F5000.0 ; Move over to prevent blob squish
```
